### PR TITLE
[docs] Task guide for semantic segmentation with LoRA

### PR DESCRIPTION
**Pull request by MKhalusova**
Moves LoRA for semantic segmentation [notebook](https://github.com/huggingface/peft/blob/main/examples/semantic_segmentation/semantic_segmentation_peft_lora.ipynb) into the docs.

### CODE REVIEW

File: docs/source/task_guides/semantic_segmentation_lora.mdx [hunk 1]
```diff
@@ -0,0 +1,445 @@
...
+classifier parameters are also trainable, and they will be serialized alongside the LoRA trainable parameters when we 
+use utility functions like `save_pretrained()` and `push_to_hub()`.
+
+Let's review the rest of the parameters:
+
+- `r`: The dimension used by the LoRA update matrices.
+- `alpha`: Scaling factor.
+- `bias`: Specifies if the `bias` parameters should be trained. `None` denotes none of the `bias` parameters will be trained.
+
+`r` and `alpha` together control the total number of final trainable parameters when using LoRA, giving you the flexibility
```
Comment (pacman100) [inline docs/source/task_guides/semantic_segmentation_lora.mdx:279]:
```suggestion
```

Comment (pacman100) [reply docs/source/task_guides/semantic_segmentation_lora.mdx:279]: removing this line as it isn't correct. 
Actually, the delta activations from LoRA are scaled as per alpha/r. So, higher alpha means that higher weightage is given to the LoRA activations and as such the gradients are also scaled accordingly. MAM paper uses LoRA and Scaled Parallel Adapter with alpha being 4X compared to r. The LoRA paper itself uses alpha ranging from 1X-8X that of r.

Alpha doesn't change the number of trainable parameters, rather functions as a scaling factor.

File: docs/source/task_guides/semantic_segmentation_lora.mdx [hunk 2]
```diff
@@ -0,0 +1,445 @@
...
+are trainable, while the pre-trained parameters and the randomly initialized classifier parameters are kept frozen. 
+However, we do want to train the classifier parameters. By specifying the `modules_to_save` argument, we ensure that the 
+classifier parameters are also trainable, and they will be serialized alongside the LoRA trainable parameters when we 
+use utility functions like `save_pretrained()` and `push_to_hub()`.
+
+Let's review the rest of the parameters:
+
+- `r`: The dimension used by the LoRA update matrices.
+- `alpha`: Scaling factor.
+- `bias`: Specifies if the `bias` parameters should be trained. `None` denotes none of the `bias` parameters will be trained.
```
Comment (pacman100) [inline docs/source/task_guides/semantic_segmentation_lora.mdx:277]: insightful and easy to follow!

**Review by pacman100 — APPROVED** [review]
Thank you @MKhalusova for this nice usage guide on semantic segmentation using lora! ✨
Left a comment.

cc @sayakpaul for a quick look as he had worked on this example

File: docs/source/task_guides/semantic_segmentation_lora.mdx (hunk 1 above)
Comment (MKhalusova) [reply docs/source/task_guides/semantic_segmentation_lora.mdx:279]: Thank you for the explanation!

**Pull request merged**